### PR TITLE
Adding Bubblegum guide for hashed NFT data

### PR DESCRIPTION
**Pull request by danenbm**
Link to preview of specific updated page: https://developer-hub-git-danenbm-hashed-nft-data-metaplex.vercel.app/bubblegum/hashed-nft-data

### CODE REVIEW

File: src/pages/bubblegum/hashed-nft-data.md [hunk 1]
```diff
@@ -4,4 +4,119 @@ metaTitle: Bubblegum - Hashing NFT Data
...
+    pub is_mutable: bool,
+    /// nonce for easy calculation of editions, if present
+    pub edition_nonce: Option<u8>,
+    /// Since we cannot easily change Metadata, we add the new DataV2 fields here at the end.
+    pub token_standard: Option<TokenStandard>,
+    /// Collection
+    pub collection: Option<Collection>,
+    /// Uses
+    pub uses: Option<Uses>,
+    pub token_program_version: TokenProgramVersion,
```
Comment (febo) [inline src/pages/bubblegum/hashed-nft-data.md:31]: Can we add comments here?

File: src/pages/bubblegum/hashed-nft-data.md [hunk 2]
```diff
@@ -4,4 +4,119 @@ metaTitle: Bubblegum - Hashing NFT Data
...
+    pub creators: Vec<Creator>,
```
Comment (febo) [inline src/pages/bubblegum/hashed-nft-data.md:32]: And here as well? This is cosmetic, but all other fields have comments apart from the last 2.

File: src/pages/bubblegum/hashed-nft-data.md [hunk 3]
```diff
@@ -4,4 +4,119 @@ metaTitle: Bubblegum - Hashing NFT Data
...
+}
+```
+
+First the metadata is hashed, using the keccak256 hash function.  This hashNote that the metadata is hashed, and then hashed again with the `seller_fee_basis_points`.  This makes it easier for marketplaces to validate seller fee basis points.
```
Comment (febo) [inline src/pages/bubblegum/hashed-nft-data.md:36]: Typo -> "This hashNote"

Does it make sense to elaborate why it is easier?

**Review by febo — CHANGES_REQUESTED** [review]
Minor comments.

File: src/pages/bubblegum/hashed-nft-data.md (hunk 1 above)
Comment (danenbm) [reply src/pages/bubblegum/hashed-nft-data.md:32]: Done

File: src/pages/bubblegum/hashed-nft-data.md (hunk 2 above)
Comment (danenbm) [reply src/pages/bubblegum/hashed-nft-data.md:34]: Done

File: src/pages/bubblegum/hashed-nft-data.md (hunk 3 above)
Comment (danenbm) [reply src/pages/bubblegum/hashed-nft-data.md]: Added explanations.

**Review by MarkSackerberg — APPROVED** [review]

File: src/pages/bubblegum/hashed-nft-data.md [hunk 4]
```diff
@@ -4,4 +4,124 @@ metaTitle: Bubblegum - Hashing NFT Data
...
+    /// Which token program version (currently only `TokenProgramVersion::Original`` is supported).
+    pub token_program_version: TokenProgramVersion,
+    /// The array of creators of the cNFT.
+    pub creators: Vec<Creator>,
+}
+```
+
+First the metadata is hashed, using the keccak-256 hash function.  Keccak-256 is much stronger than SHA-256, and is used in Solana as well as other blockchains such as Ethereum.
+
+Note that the metadata is hashed, and then hashed again with the `seller_fee_basis_points`.  This makes it easier for marketplaces to validate seller fee basis points, because they do not have to pass a full `MetadataArgs` struct around (which can be up to 457 bytes in length).  Instead, they can pass a 32-byte array of already-hashed `MetadataArgs`, and the `u16` `seller_fee_basis_points`, and by hashing them together they can recreate the data hash.
```
Comment (febo) [inline src/pages/bubblegum/hashed-nft-data.md:40]: Suggestion: `u16` (2 bytes) `seller_fee_basis_points`

**Review by febo — APPROVED** [review]
LGTM - just left one suggestion.

File: src/pages/bubblegum/hashed-nft-data.md (hunk 4 above)
Comment (danenbm) [reply src/pages/bubblegum/hashed-nft-data.md]: done

**Review by blockiosaurus — APPROVED** [review]

**Review by febo — APPROVED** [review]

**Pull request merged**